lib/variable/inverse.cpp

```diff
@@ -0,0 +1,35 @@
+// SPDX-License-Identifier: BSD-3-Clause
+// Copyright (c) 2023 Scipp contributors (https://github.com/scipp)
+#include "scipp/variable/inverse.h"
+
+#include "scipp/common/overloaded.h"
+#include "scipp/core/element/arg_list.h"
+#include "scipp/core/spatial_transforms.h"
+#include "scipp/core/transform_common.h"
+
+#include "scipp/variable/transform.h"
+
+namespace scipp::variable {
+namespace element {
+constexpr auto inverse =
+    overloaded{core::element::arg_list<Eigen::Matrix3d, Eigen::Affine3d,
+                                       core::Translation, core::Quaternion>,
+               core::transform_flags::expect_no_variance_arg<0>,
+               core::transform_flags::expect_no_variance_arg<1>,
+               [](const auto &transform) { return transform.inverse(); },
+               [](const units::Unit &u) {
+                 // This is not correct for linear transforms (Eigen::Matrix3d).
+                 // But that case is handled separately by the caller.
```

Comment:
Isn't it wrong also for quaternions?

Reply:
Quaternions are always dimensionless (unless people assign a unit to them which scipp does not prevent but makes the transform ill-defined). So the kernel does the right thing.

Reply:
Why is the transform ill-defined if not dimensionless?

Reply:
So why is it supported by linear transforms?

And as you point out, rotations *can* have a unit, so I think we should handle it correctly, by using the inverse, such as $R^{-1}Rv=v$

Reply:
What I pointed out is that this is possible:
```python
rot = sc.spatial.rotation(...)
rot.unit = 'm'
```
But this is nonsensical because rotations preserve lengths. (Note that $R^{-1} = R^T$. This would not be the case with a unit.)

Rotations are only a special case of linear transformations, though. The latter can have a unit because it can scale a vector arbitrarily.

Reply:
I am not saying that it makes sense, it it seems that you are suggesting that because it is wrong, introducing a second problem is not an issue? I would argue that we should handle it as well as possible (by using the inverse of the unit, which is no more complicated to implement that what you have).

The other alternative is to prevent setting a unit in this case.